File: Python/RSAEncryption/math_operation/primeNumberOp.py

import random

# Function to check is the number that given through the parameter is a prime number or not
# return value in boolean
def isPrime(number):
    if number < 2:
        return False
    
    i = 2

    while i < number // 2 + 1:
        if number%i == 0:
            return False
        i = i + 1

    return True


# Function to generate prime number from the given range from the parameter
# return value in number format like INT
def generatePrimeNumber(min, max):
    generated = random.randint(min, max)

    while not isPrime(generated):
        generated = random.randint(min, max)
    
    return generated

File: PHP/RSAEncryption/lib/math/prime-number-op.php
/*
    HANDLE Prime Number Operation
*/

<?php 

    /*
        Function to check if the number given in the parameter is a prime number or not
        Return boolean
    */
    function isPrime($number){
        if($number < 2){
            return false;
        }

        for($i = 0; $i < ($number/2) + 1; $i++){
            if($number % $i === 0){
                return false;
            }
        }

        return true;
    }

    /*
        Function to generate random prime number by the range given in the min and max parameter
        Return integer
    */
    function generatePrime($min, $max){
        $primeNumber = rand($min, $max);

        while(!isPrime($primeNumber)){
            $primeNumber = rand($min, $max);
        }

        return $primeNumber;
    }

?>